src/tuesday/core/summaries/powerspectra.py

```diff
@@ -13,58 +18,121 @@
 from scipy.interpolate import RegularGridInterpolator
 
 
-def calculate_ps(  # noqa: C901
-    lc,
-    lc_redshifts,
-    box_length,
-    box_side_shape=None,
-    zs=None,
-    chunk_size=None,
-    chunk_skip=37,
-    calc_2d=True,
-    nbins=50,
-    k_weights=ignore_zero_ki,
-    postprocess=True,
-    kpar_bins=None,
-    log_bins=True,
-    crop=None,
-    calc_1d=False,
-    nbins_1d=14,
-    calc_global=False,
-    mu=None,
-    bin_ave=True,
-    interp=None,
-    prefactor_fnc=power2delta,
-    interp_points_generator=None,
-    get_variance=False,
+@dataclass(frozen=True)
+class PowerSpectrum:
```

Comment:
Have a `SphericalPS` and `CylindricalPS` class -- don't bunch them into one god-object.